python/dask_cudf/dask_cudf/io/csv.py

```diff
@@ -106,7 +106,9 @@ def read_csv_without_chunksize(path, **kwargs):
 
     name = "read-csv-" + tokenize(path, **kwargs)
 
-    meta = cudf.read_csv(filenames[0], **kwargs)
+    # Read "head" of first file (first 5 rows).
+    # Convert to empty df for metadata.
+    meta = cudf.read_csv(filenames[0], nrows=5, **kwargs).iloc[:0]
```

Comment:
Just a note that this will slice to zero size but will still keep some memory around due to the slice being a zero copy reference. I don't think it's problematic in this instance though.

Reply:
Yeah, that makes sense - There is actually no need to do the slicing here, since Dask will do this within  `new_dd_object` (in `make_meta`), but I thought it made sense to be explicit.